docs/architecture/architecture.md

```diff
@@ -0,0 +1,674 @@
+# Clear Containers Architecture
+
+* [Overview](#overview)
+    * [Hypervisor](#hypervisor)
+    * [Agent](#agent)
+    * [Runtime](#runtime)
+        * [Configuration](#configuration)
+        * [Significant commands](#significant-commands)
+            * [create](#create)
+            * [start](#start)
+            * [exec](#exec)
+            * [kill](#kill)
+            * [delete](#delete)
+    * [Proxy](#proxy)
+    * [Shim](#shim)
+    * [Networking](#networking)
+    * [Storage](#storage)
+* [Appendices](#appendices)
+    * [DAX](#dax)
+    * [Previous Releases](#previous-releases)
+    * [Resources](#resources)
+
+## Overview
+
+This is an architectural overview of Clear Containers, based on the 3.0 release.
+
+The [Clear Containers runtime (cc-runtime)](https://github.com/clearcontainers/runtime)
+is compatible with the [OCI](https://github.com/opencontainers) [runtime specification](https://github.com/opencontainers/runtime-spec)
+and therefore works seamlessly with the
+[Docker\* Engine](https://www.docker.com/products/docker-engine) pluggable runtime
+architecture. It also supports the [Kubernetes\* Container Runtime Interface (CRI)](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/apis/cri/v1alpha1/runtime)
+through the [CRI-O\*](https://github.com/kubernetes-incubator/cri-o) implementation.
+In other words, you can transparently select between the
+[default Docker and CRI-O runtime (runc)](https://github.com/opencontainers/runc)
+and `cc-runtime`.
+
+![Runtime and virtcontainers](arch-images/runtime-vc-relationship.png)
```

Comment:
This .png isn't included in the PR or has a different name. It will bring a 404 error.

Reply:
y'ouch - git blame says that wa sfrom me, but I don't recall adding this, nor can I find this image anywehere... removing now...